File: app/src/main/java/com/akilimo/mobile/Akilimo.java
package com.akilimo.mobile;


import androidx.appcompat.app.AppCompatDelegate;
import androidx.multidex.MultiDexApplication;

import com.blongho.country_data.World;


import net.danlew.android.joda.JodaTimeAndroid;

import java.util.Locale;

import dev.b3nedikt.app_locale.AppLocale;
import dev.b3nedikt.app_locale.SharedPrefsAppLocaleRepository;
import dev.b3nedikt.reword.RewordInterceptor;
import io.github.inflationx.viewpump.ViewPump;

public class Akilimo extends MultiDexApplication {

    private String LOG_TAG = Akilimo.class.getSimpleName();

    @Override
    public void onCreate() {
        super.onCreate();

        AppLocale.setSupportedLocales(Locales.APP_LOCALES);
        SharedPrefsAppLocaleRepository prefs = new SharedPrefsAppLocaleRepository(this);

        AppLocale.setAppLocaleRepository(prefs); //persist changes
        Locale desiredLocale = prefs.getDesiredLocale();
        if (desiredLocale != null) {
            AppLocale.setDesiredLocale(desiredLocale);
        }

        ViewPump.init(ViewPump.builder()
                .addInterceptor(RewordInterceptor.INSTANCE)
                .build()
        );
        //@FIX This is used to enable proper vector support for android 4.4 and below, it causes crashing when firing up the application
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
        JodaTimeAndroid.init(this);
        World.init(this);
    }
}

File: app/src/main/java/com/akilimo/mobile/Akilimo.kt
package com.akilimo.mobile

import androidx.appcompat.app.AppCompatDelegate
import androidx.multidex.MultiDexApplication
import com.akilimo.mobile.rest.retrofit.RetrofitManager
import com.akilimo.mobile.utils.PreferenceManager
import com.blongho.country_data.World
import com.jakewharton.threetenabp.AndroidThreeTen


class Akilimo : MultiDexApplication() {
    override fun onCreate() {
        super.onCreate()
        // TODO: Revise and remove in future releases
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)


        val sessionManager = PreferenceManager(this@Akilimo)
        val akilimoEndpoint = sessionManager.akilimoEndpoint
        val fuelrodEndpoint = sessionManager.fuelrodEndpoint

        RetrofitManager.init(this@Akilimo, akilimoEndpoint, fuelrodEndpoint)

        //@FIX This is used to enable proper vector support for android 4.4 and below, it causes crashing when firing up the application
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
        AndroidThreeTen.init(this@Akilimo)
        World.init(this@Akilimo)
    }
}
